disorder-core/src/Disorder/Core/NaturalInt.hs

```diff
@@ -0,0 +1,15 @@
+module Disorder.Core.NaturalInt (
+    NaturalInt (..)
+  ) where
+
+import           Control.Applicative
+
+import           Test.QuickCheck
+
+
+data NaturalInt = NaturalInt {
+    value :: Int
+  } deriving (Eq, Show)
+
+instance Arbitrary NaturalInt where
+  arbitrary = NaturalInt <$> (suchThat arbitrary ((<=) 0))
```

Comment:
If you defined an `Integral` and `Bounded` instance for these you could just use `arbitrary = arbitrarySized Bounded Integral`, [doc here] (https://hackage.haskell.org/package/QuickCheck-2.4.0.1/docs/Test-QuickCheck.html#v:arbitrarySizedBoundedIntegral)

`suchThat` effectively does a filter, while the above, or `choose` will only select from valid elements